toucan_connectors/awsathena/awsathena_connector.py

```diff
@@ -0,0 +1,51 @@
+import awswrangler as wr
+import boto3
+import pandas as pd
+from pydantic import Field, SecretStr, constr
+
+from toucan_connectors.toucan_connector import ToucanConnector, ToucanDataSource
+
+
+class AwsathenaDataSource(ToucanDataSource):
+    name: str = Field(..., description='Your AWS Athena connector name')
+    database: constr(min_length=1) = Field(
+        ..., description='The name of the database you want to query.'
+    )
+    query: constr(min_length=1) = Field(
+        ...,
+        description='The SQL query to execute.',
+        widget='sql',
+    )
+
+
+class AwsathenaConnector(ToucanConnector):
+    data_source_model: AwsathenaDataSource
+
+    name: str = Field(..., description='Your AWS Athena connector name')
+
+    s3_output_bucket: str = Field(
+        ..., description='Your S3 Output bucket (where query results are stored.)'
+    )
+    aws_access_key_id: str = Field(..., description='Your AWS access key ID')
+    aws_secret_access_key: SecretStr = Field(None, description='Your AWS secret key')
+    region_name: str = Field(..., description='Your AWS region name')
+
+    def get_session(self):
+        session = boto3.Session(
+            aws_access_key_id=self.aws_access_key_id,
+            # This is required because this gets appended by boto3
+            # internally, and a SecretStr can't be appended to an str
+            aws_secret_access_key=self.aws_secret_access_key.get_secret_value(),
+            region_name=self.region_name,
+        )
+        return session
```

Comment:
```suggestion
        return boto3.Session(
            aws_access_key_id=self.aws_access_key_id,
            # This is required because this gets appended by boto3
            # internally, and a SecretStr can't be appended to an str
            aws_secret_access_key=self.aws_secret_access_key.get_secret_value(),
            region_name=self.region_name,
        )
```